app/webpacker/components/Live/Admin/DoubleCheck/index.jsx

```diff
@@ -0,0 +1,151 @@
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
+import {
+  Button, Card, Form,
+  Grid, Header,
+  Segment,
+} from 'semantic-ui-react';
+import { useMutation, useQuery } from '@tanstack/react-query';
+import { events } from '../../../../lib/wca-data.js.erb';
+import WCAQueryClientProvider from '../../../../lib/providers/WCAQueryClientProvider';
+import AttemptResultField from '../../../EditResult/WCALive/AttemptResultField/AttemptResultField';
+import updateRoundResults from '../../api/updateRoundResults';
+import getRoundResults from '../../api/getRoundResults';
+import Loading from '../../../Requests/Loading';
+
+export default function Wrapper({
+  competitionId, competitors, round,
+}) {
+  return (
+    <WCAQueryClientProvider>
+      <Competitors
+        eventId={round.event.id}
+        competitionId={competitionId}
+        competitors={competitors}
+        round={round}
+      />
+    </WCAQueryClientProvider>
+  );
+}
+
+function zeroedArrayOfSize(size) {
+  return Array(size).fill(0);
+}
+
+function Competitors({
+  competitionId, competitors, round, eventId,
+}) {
+  const event = events.byId[eventId];
+  const solveCount = event.recommendedFormat().expectedSolveCount;
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const [registrationId, setRegistrationId] = useState(competitors[0].id);
+  const [attempts, setAttempts] = useState(zeroedArrayOfSize(solveCount));
+
+  const { data: results, isLoading } = useQuery({
+    queryKey: [round.id, 'results'],
+    queryFn: () => getRoundResults(round.id, competitionId),
+  });
+
+  const {
+    mutate: mutateUpdate, isPending: isPendingUpdate,
+  } = useMutation({
+    mutationFn: updateRoundResults,
+  });
+
+  const handleSubmit = async () => {
+    mutateUpdate({
+      roundId: round.id, registrationId, competitionId, attempts,
+    });
+  };
+
+  const handleAttemptChange = (index, value) => {
+    const newAttempts = [...attempts];
+    newAttempts[index] = value;
+    setAttempts(newAttempts);
+  };
+
+  const handleRegistrationIdChange = useCallback((_, { value }) => {
```

Comment:
Does this need to be a callback? I think a regular const anonymous function would be fine.